### PR TITLE
Stash deletion of submodules

### PR DESCRIPTION
**Pull request by novalis**

### CODE REVIEW

File: node/lib/util/stash_util.js
Comment (leahnp) [inline node/lib/util/stash_util.js:252]: does this make a special stash, that would be different than a user made stash?

Comment (novalis) [reply node/lib/util/stash_util.js:252]: It's different in the meta repo, but not in the submodule repo.  Since regular stashes in the meta repo are not really a thing that does anything, I think this is OK.

**Comment by leahnp** [comment]
This is quite complex, I think I understand it and reviewed it the best I can - I am happy to approve unless you want another set of eyes on it.

**Review by leahnp — APPROVED** [review]

File: node/lib/util/repo_status.js
Comment (shijinglu) [inline node/lib/util/repo_status.js:376]: nit: you can use optional chaining: `?.`

File: node/lib/util/rm.js
Comment (shijinglu) [inline node/lib/util/rm.js:477]: do you want to raise exception if `("ENOTEMPTY" !== e.code)`? Otherwise the exception will be swallowed, right?

File: node/lib/util/stash_util.js
Comment (shijinglu) [inline node/lib/util/stash_util.js:150]: It calculates `treeToWorkdir`, which implies it always `includeUntracked`, right? 

How do you handle the `includeUntracked` parameter in `exports.save`?

Comment (novalis) [reply node/lib/util/stash_util.js:150]: This only applies to .gitmodules, which should never be untracked?

File: node/lib/util/repo_status.js
Comment (novalis) [reply node/lib/util/repo_status.js:376]: Not with this version of jshint :(

**Pull request merged**
